src/bun.js/test/jest.zig

```diff
@@ -3592,6 +3592,147 @@ pub const Expect = struct {
         return .zero;
     }
 
+    pub fn toIncludeRepeated(this: *Expect, globalThis: *JSGlobalObject, callFrame: *CallFrame) callconv(.C) JSValue {
+        defer this.postMatch(globalThis);
+
+        const thisValue = callFrame.this();
+        const arguments_ = callFrame.arguments(2);
+        const arguments = arguments_.ptr[0..arguments_.len];
+
+        if (arguments.len < 2) {
+            globalThis.throwInvalidArguments("toIncludeRepeated() requires 2 arguments", .{});
+            return .zero;
+        }
+
+        const substring = arguments[0];
+        substring.ensureStillAlive();
+
+        if (!substring.isString()) {
+            globalThis.throw("toIncludeRepeated() requires the first argument to be a string", .{});
+            return .zero;
+        }
+
+        const count = arguments[1];
+        count.ensureStillAlive();
+
+        if (!count.isNumber()) {
+            globalThis.throw("toIncludeRepeated() requires the second argument to be a number", .{});
+            return .zero;
+        }
+
+        const countAsNum = std.fmt.parseInt(u32, count.toString(globalThis).toSlice(globalThis, default_allocator).slice(), 10) catch {
+            globalThis.throw("toIncludeRepeated() requires the second argument to be a number", .{});
+            return .zero;
+        };
+
+        const expect_string = Expect.capturedValueGetCached(thisValue) orelse {
+            globalThis.throw("Internal consistency error: the expect(value) was garbage collected but it should not have been!", .{});
+            return .zero;
+        };
+
+        if (!expect_string.isString()) {
+            globalThis.throw("toIncludeRepeated() requires the expect(value) to be a string", .{});
+            return .zero;
+        }
+
+        const not = this.op.contains(.not);
+        var pass = std.mem.containsAtLeast(u8, expect_string.toString(globalThis).toSlice(globalThis, default_allocator).slice(), countAsNum, substring.toString(globalThis).toSlice(globalThis, default_allocator).slice());
+
+        if (not) pass = !pass;
+        if (pass) return thisValue;
+
+        var formatter = JSC.ZigConsoleClient.Formatter{ .globalThis = globalThis, .quote_strings = true };
+        const expect_string_fmt = expect_string.toFmt(globalThis, &formatter);
+        const substring_fmt = substring.toFmt(globalThis, &formatter);
+        const times_fmt = count.toFmt(globalThis, &formatter);
+
+        if (not) {
+            const expected_line = "Expected to not include: <green>{any}<r> <green>{any}<r> times\n";
+            const received_line = "Received: <red>{any}<r>\n";
+            const fmt = comptime getSignature("toIncludeRepeated", "<green>expected<r>", true) ++ "\n\n" ++ expected_line ++ received_line;
+            globalThis.throwPretty(fmt, .{ substring_fmt, times_fmt, expect_string_fmt });
+            return .zero;
+        }
+
+        const expected_line = "Expected to include: <green>{any}<r> <green>{any}<r> times \n";
```

Comment:
How about this:

* If times = 0: `Expected to not include: {}`
* If times = 1: `Expected to include: {}`
* If times > 1: `Expected to include: {} ({} times)`
* If not and times = 0: `Expected to include: {}`
* If not and times = 1: `Expected not to include: {}`
* If not and times > 1: `Expected not to include: {} ({} times)`